src/parser/mod.rs

```diff
@@ -30,21 +30,28 @@ impl Parser {
     pub fn parse_all(&mut self, end_token_type: TkType) -> Result<Vec<TopAst>> {
         let mut program = vec![];
         while self.peek(0)?.tk_type() != &end_token_type {
-            let tag = self.parse_tag()?;
-            let ast = self.parse_top_ast()?;
-            program.push(TopAst::new(tag, ast));
+            program.push(TopAst {
+                ast: self.parse_top_ast()?,
+            });
         }
         Ok(program)
     }
     pub fn parse_tag(&mut self) -> Result<Option<Tag>> {
         if self.predict_and_consume(vec![TkType::AtSign]).is_ok() {
             let tag_name = self.parse_identifier()?;
-            Ok(Some(Tag::new(tag_name)))
+            let properties = self.parse_many_if_has_open_token(
+                TkType::OpenParen,
+                TkType::CloseParen,
+                TkType::Comma,
+                |parser| Ok(parser.parse_identifier()?),
+            )?;
```

Comment:
`parse_many_if_has_open_token` as its name, it only parses when matched the open token